test/call/github-events-to-s3.test.ts

```diff
@@ -60,16 +50,35 @@ describe('githubEventsToS3', () => {
   it('should upload to S3 on event listened', async () => {
     mockS3Client.send.mockResolvedValue({});
 
-    await githubEventsToS3(app, context, resource);
+    await githubEventsToS3(app, context);
 
     expect(mockS3Client.send).toHaveBeenCalledWith(expect.any(PutObjectCommand));
     expect(app.log.info).toHaveBeenCalledWith('GitHub Event uploaded to S3 successfully.');
   });
 
+  it('should not upload to S3 on event listened on private repo', async () => {
+    context = {
+      name: 'name',
+      id: 'id',
+      payload: {
+        repository: {
+          name: 'repo',
+          owner: { login: 'org' },
+          private: true,
+        },
+      },
+    };
+    mockS3Client.send.mockResolvedValue({});
+
+    await githubEventsToS3(app, context);
+
+    expect(mockS3Client.send).not.toHaveBeenCalledWith(expect.any(PutObjectCommand));
```

Comment:
If you add an app.log.error for private repo then you can check if that is being called when repo is private.